### ISSUE TITLE
Other R object is outputed in addition to the code sent from a script

### ISSUE THREAD

**Issue by acid-bbq** [emacs-ess/ESS#1138]
My config is:

Windows 10
GNU Emacs 27.1
ESS 18.10.3snapshot [elpa: 20210331.804]
Org mode 9.4.4
R 3.6.3

I first do the following:
> 1. I open an org file
> 2. I hit `C-c '` to open a chunk of code. It opens it in a separate buffer
> 3. In this new buffer ESS[R], I send code with `C-c C-n` or `C-c C-r` or `C-RET`
> 4. I am asked for R starting project directory. I hit `RET`.
> 5. Now the R console opens in an iESS [R] buffer

So I have my R script in one buffer and the R console in a second buffer.

After running some code and creating objects (e.g. vectors, data frames), when I send code from the R script, that is the buffer opened in step 3 above, the code is sent with its result, but after, I have one random object I created before which is also always shown in the R console. It doesn't happen if I run code directly from the  R console.

For example if I send this from the buffer with the R script to the R console: 
> 2+2,

It will display in the R console:
> \> 2+2
> [1] 4
>   \>[1] 0.1 0.2 0.3 0.4
> \>

where 0.1 0.2 0.3 0.4 is an object I previously created, e.g. `x <- seq(0.1, 0.4, 0.1)`. It could be this object or any other object which I created previously.

**Comment by acid-bbq** [comment]
It becomes a little bit wilder now. Not sure if it is related, but from time to time, when I select some text in my R script (by double clicking with the mouse), I also have an previously created object outputed in the console. But might also be related to previous code I sent for which the object was not displaed.

Also in the "Messages" buffer, I have this written:
```
Mark set
eldoc error: (error Buffer  *ess-command-output* has no process)
```
But the error is not always popping up when I double click, I cannot figure out when it comes. When I send code from the script this message does not appear.

The annoying thing is that it is not always a vector as stated above. It can be for example, a long dataframe. So it can completely mess it up with the actual output I want to check.

**Comment by mmaechler** [comment]
Well, just a heads up:  I've also only recently (last 1-3 weeks?) seen very occasionally that  "eval-region" or similar evals have produced complete havoc in my `*R*` R console buffer.   I've not been able to reproduce this easily, nor notice under which circumstances it happens,  but *when* it happens it's very unpleasant... had confused me initially quite a bit...  ((I know this all is not so useful, unless we find reproducible situations))

**Comment by lionel-** [comment]
Unfortunately this sort of problems is likely not reproducible in natural situations. They are probably related to timing issues with background commands in the process filter. When this happens, it would be useful to copy the messed output in the process buffer as well as the contents of the ` *ess-command-output*` buffer, and a description of what you were doing (e.g. eval-region).

**Comment by acid-bbq** [comment]
> it would be useful to copy the messed output in the process buffer as well as the contents of the *ess-command-output* buffer

This goes beyond my abilities... What do you mean by "messed output"? The object which is displayed? What is the process buffer? How to open the `*ess-command-output*`? But I'll be happy to do it if you could let me know how to do it.

**Comment by lionel-** [comment]
> What do you mean by "messed output"? The object which is displayed?

I guess so

> What is the process buffer?

The buffer where "objects are displayed". I.e. the buffer where output from the R process is displayed.

> How to open the *ess-command-output*?

Just like any other buffer. It is hidden from the buffer list because its name starts with an empty space, so if you look for buffers starting with `" "` you should be able to find it.

**Comment by acid-bbq** [comment]
Ok. So don't know whether that's of great interest

That's for instance, the head of a data frame I have in the `*R*` buffer which is outputed after I send code from the `*ESS[R]* ` buffer (i.e. the script). When I send code with `C-RET`, *ess-command-output* is empty.

Difficult to say when it first occurred, I only used `C-RET`, line-by-line and by region. I think it happened when I send a region (but not the first time I did it). Before this I managed to send let's say at least 100-200 lines of code.
```
>                                             0 1 Total Total
1010244.5883873.2019-08-25.2019-08-28.A.1.3 1 0     1     2
1010467.5734636.2019-03-25.2019-03-30.A.1.4 1 0     1     2
1010489.5586450.2018-11-12.2018-11-17.A.1.1 1 0     1     2
1010525.5615386.2018-12-11.2018-12-12.B.1.3 1 0     1     2
1010525.5615386.2018-12-18.2018-12-20.B.1.3 0 1     1     2
1010525.5693478.2019-02-17.2019-02-19.A.1.4 1 0     1     2
```

The same was also outputed without sending code, I was writing something in the script, and it shown. Here however, the `*ess-command-output*` displayed briefly `ess-output-delimiter75-START` or `ess-output-delimiter75-START`, but for sure with another number instead of 75. Then it was over written by:
`(list "base" '(("x" . "") ("subset" . "") ("..." . "")) '("x" "..." "subset" "select" "drop"))`
This happen when I was typing this `test <- subset(temp,)`. I  really did not complete the writing.

Please let me know whether it helps, or what esle I can do.

**Comment by acid-bbq** [comment]
Next level issue: (This one will really prevent working properly...)

Looks like that instead of wanting to print a previous object, it wants to execute, I suppose, a loop I executed and took some time to be completed (10-20 sec).

I have this in my *R* buffer:

`>`

Now I just have to click somewhere in my script (or run a line of code) and then it displays:

`>+`

After some time (10-20 sec, don't know if time is related to the loop time or if this issue is really linked to the loop...) I can hit `RET` on there and it goes to a next line with `>`

And in `Messages` buffer I have:
`Error running timer ‘ess--idle-timer-function’: (error "Buffer  *ess-get-words* has no process")`

**Comment by acid-bbq** [comment]
Some updates:

I've tried without loading my init.el file. (Actually I've just renamed the .emacs.d folder, so that it cannot load init.el)

Now, after running some code, when I am in my R script buffer (opened with `C-c'` within org-mode), and when I move the cursor over some lines of code the following occurs:

If the cursor goes over this:
`df.all <- rbind(unique(df1[,c("column1", "column2")]), unique(df2[,c("column1", "column2")]))`
The R console shows:
`(list "mice" '(("..." . "")) '("..." "deparse.level" "make.row.names" "stringsAsFactors" "factor.exclude" "size" "z" "rename"))`


If the cursor goes over this:
`diff.df1.df2 <- c(setdiff(colnames(df1), colnames(df2)),  setdiff(colnames(df2), colnames(df1)))`
The R console shows:
`(list "" '(("..." . "")) '("recursive" "use.names" "..." "x" "x.same" "y.same" "layout" "merge.legends"))
`

Or for example:
`levels(df$var1) <- c("A", "B", "C", "D")`
It shows:
`(list "" '(("..." . "")) '("recursive" "use.names" "..." "x" "x.same" "y.same" "layout" "merge.legends"))`

Looks to be similar to what can be shown in `*ess-command-output*`, but `*ess-command-output*` remains empty. In addition, these outputs are not showing for every lines I move the cursor onto.

**Comment by lionel-** [comment]
That means you're using the last released version of ESS from 2018.

**Comment by acid-bbq** [comment]
True, I've just tried to install the version of Vincent Goulet and it includes Emacs 27.2 and ESS 18.10.2.

I'll try the same procedure with my config including the ESS dev version.

But in the meantime, is there a previous version I can install to prevent this from happening?

**Comment by lionel-** [comment]
There is no released version of ESS compatible with Emacs 27. You need to downgrade Emacs to version 25 or 26 or use the dev version of ESS.

**Comment by acid-bbq** [comment]
Thank you for all your swift replies. They are very helpful. I've downgraded to Emacs 26.3, still with the dev version of ESS and no more issues.

**Comment by r2evans** [comment]
@lionel- I can't find documentation suggesting that emacs-27 is incompatible with dev-ESS. Is this still the case? If so, is it listed somewhere? I recognize this is just for the dev-version, but ... I just upgraded to 27 (I thought it might be good in some ways), and seeing this makes me a little nervous.

**Comment by lionel-** [comment]
emacs-27 is not compatible with _released_ ESS. Dev ESS works mostly fine ("mostly" is why we haven't released yet).
